usaspending_api/awards/models.py

```diff
@@ -310,6 +328,24 @@ def get_default_fields(path=None):
             "assistance_data"  # must match related_name in TransactionAssistance
         ]
 
+    @classmethod
+    def get_or_create(cls, **kwargs):
+        """Gets and updates, or creates, a Transaction
+
+        Transactions must be unique on Award, Awarding Agency, and Mod Number
+        """
+        transaction = cls.objects.filter(
+            award=kwargs.get('award'),
+            awarding_agency=kwargs.get('awarding_agency'),
+            modification_number=kwargs.get('modification_number')
+        ).first()
```

Comment:
OK, I'll sort the records, and write a ticket for a potential unique constraint.